sycl/doc/extensions/proposed/sycl_ext_oneapi_composite_device.asciidoc

```diff
@@ -0,0 +1,290 @@
+};
+
+} // namespace sycl
+```
+
+[width="100%",%header,cols="50%,50%"]
+|===
+|Aspect
+|Description
+
+|`ext_oneapi_is_composite`
+|Indicates that the device is a composite device.  Any device with this aspect
+ will have at least two constituent component devices.
+
+|`ext_oneapi_is_component`
+|Indicates that the device is a component device of some other composite
+ device.  This applies only to a root device that is a direct component of
+ some composite device.  A sub-device will not have this aspect even if its
+ parent is a component device.
+|===
+
+=== Impact on "descendent device"
+
+This extension augments the definition of the term "descendent device" from the
+core SYCL specification as follows.  Given some composite device _C_ and some
+component of that composite device _R_, device _R_ is a descendent device of
+_C_ and all descendent devices of _R_ are also descendent devices of _C_.
+
+This definition means that a SYCL context that contains a composite device is
+compatible with any of its component devices, even if those component devices
+are not contained by the context.  See the core SYCL specification for details.
+
+
+== Impact to the ONEAPI_DEVICE_SELECTOR
+
+The `ONEAPI_DEVICE_SELECTOR` is an environment variable that is specific to the
+{dpcpp} implementation.  Therefore, this section that describes the interaction
+between this extension and that environment variable is non-normative and does
+not apply to other SYCL implementations that may support this extension.
+
+The `ONEAPI_DEVICE_SELECTOR` environment variable determines the list of root
+devices that are returned from `device::get_devices`: _R0_, _R1_, _R2_, etc.
+The devices returned from `get_composite_devices` are computed from this list
+by iterating over the elements _Ri_ in order:
+
+* Start with an empty list of composite devices.
+* If device _Ri_ is a component of some composite device _C_ and if all other
+  components of _C_ are also in the list of root devices returned from
+  `device::get_devices`, then _C_ is appended to the list of composite devices
+  unless _C_ is already in that list.
+
+This algorithm ensures that a composite device is made visible to the
+application only if all of its components are also visible.  This is important
+for two reasons.  The first reason is philosophical.  We do not want to expose
+a composite device unless it has at least two component devices.  This is
+similar to our decision in the core SYCL spec to disallow partitioning into
+sub-devices which results in only a single sub-device.  We don't want to expose
+partitioning when the parent and child represent exactly the same hardware.
+The second reason is practical.  The {dpcpp} implementation will associate
+each composite device with a single Level Zero native device handle, which
+represents the entire card.  There is no way to get a Level Zero handle that
+represents a subset of the tiles.  Therefore, we do not expose a component
```

Comment:
```suggestion
represents a subset of the tiles.  Therefore, we do not expose a composite
```

I think this should be composite, right?

Reply:
Yes, thanks for catching this!